### PR TITLE
fix(TBD-11968):CVE commons-collections 3.2.1

### PR DESCRIPTION
**Pull request by lbourgeois**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml [hunk 1]
```diff
@@ -1834,9 +1834,8 @@
 
         <libraryNeeded
             context="plugin:org.talend.libraries.apache.common"
-            id="commons-collections-3.2.1.jar"
-            name="commons-collections-3.2.1.jar" mvn_uri="mvn:org.talend.libraries/commons-collections-3.2.1/6.0.0"
-            uripath="platform:/plugin/org.talend.libraries.apache.common/lib/commons-collections-3.2.1.jar"
```
Comment (ikulish) [inline main/plugins/org.talend.designer.components.localprovider/plugin.xml:1839]: why there is no uripath for new lib?

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml [hunk 2]
```diff
@@ -2483,10 +2482,9 @@
             uripath="platform:/plugin/org.talend.libraries.apache.common/lib/commons-collections4-4.0.jar">
         </libraryNeeded>
         <libraryNeeded
-            id="commons-collections-3.2.1.jar"
-            name="commons-collections-3.2.1.jar" mvn_uri="mvn:org.talend.libraries/commons-collections-3.2.1/6.0.0"
-            context="plugin:org.talend.libraries.apache.common"
-            uripath="platform:/plugin/org.talend.libraries.apache.common/lib/commons-collections-3.2.1.jar">
```
Comment (ikulish) [inline main/plugins/org.talend.designer.components.localprovider/plugin.xml:2489]: why there is no uripath for new lib?

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml [hunk 3]
```diff
@@ -3065,7 +3063,6 @@
                 name="HDFS-LIB-HDP_2_3">
             <library id="avro-1.7.5.jar"/>
             <library id="commons-cli-1.2.jar"/>
-            <library id="commons-collections-3.2.1.jar"/>
```
Comment (ikulish) [inline main/plugins/org.talend.designer.components.localprovider/plugin.xml:3068]: why did you remove it instead of updating to version 3.2.2?

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml (hunk 1 above)
Comment (lbourgeois) [reply main/plugins/org.talend.designer.components.localprovider/plugin.xml:1839]: uripath is not needed anymore, it is an heritage from time when jars where directly included in Studio and maven not used.
Now only needed part is mvn_uri

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml (hunk 2 above)
Comment (lbourgeois) [reply main/plugins/org.talend.designer.components.localprovider/plugin.xml:2489]: uripath is not needed anymore, it is an heritage from time when jars where directly included in Studio and maven not used.
Now only needed part is mvn_uri

File: main/plugins/org.talend.designer.components.localprovider/plugin.xml (hunk 3 above)
Comment (lbourgeois) [reply main/plugins/org.talend.designer.components.localprovider/plugin.xml:3068]: I removed because it is part of a module group used for HDP 2.3 which is not anymore in Studio so this is dead code that should be cleaned more globally.

**Review by mtuhai — APPROVED** [review]

**Review by sponomarova — APPROVED** [review]
LGTM

**Review by firefryy — APPROVED** [review]
lgtm
